new_webapp/src/actions/index.js

```diff
@@ -165,4 +166,21 @@ export const getMoreExamples = ({candidateId, code, usages}) => (dispatch, getSt
                 message: errorResult.error || "Unknown Error",
             }));
         })
-}
\ No newline at end of file
+}
+
+// This is where a request needs to be sent to the server
+// id: uuid
+export const doStop = ({id}) => (dispatch) => {
+    Search.sendStopSignal({id})
+    .then(result => {
+			  dispatch(setSearchStatus({status:DONE}));
```

Comment:
Shouldn't you set the search status to DONE in all cases?
nit pick: the indentation is a bit wonky. I've been using 4 space indent.